aiida/repository/control.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for overall repository control commands."""
+import typing
+
+
+def repository_maintain(live: bool = False, pass_down: typing.Optional[str] = None) -> None:
+    """Performs maintenance tasks on the repository."""
+    from aiida import orm
+    from aiida.manage.manager import get_manager
+    from aiida.repository import Repository
+
+    if not live:
+        print('Here I should be trying to lock down the profile...')
+
+    manager = get_manager()
+    repository = manager.get_backend().get_repository()
+
+    # Propagate the deleted nodes to the repository backend
+    keyset_existing = set(repository.list_objects())
+
+    keyset_recorded = set()
+    query = orm.QueryBuilder()
+    query.append(orm.Node, project=['repository_metadata'])
+    for metadata, in query.iterall():
+        node_repo = Repository.from_serialized(backend=repository, serialized=metadata)
+        keyset_recorded.update(node_repo.get_file_keys())
```

Comment:
this can all be replaced by:

```suggestion
   keyset_recorded = set(orm.Node.objects(backend).iter_repo_keys(batch_size=batch_size))
```